File: Hard/regular_expression_matching.js
/**
 * @param {string} s
 * @param {string} p
 * @return {boolean}
 */
var isMatch = function(s, p) {
    const regx = new RegExp(`^${p}$`);
    return regx.test(s);
};

console.log(isMatch("aa", "a")); //False
console.log(isMatch("aa", "a*")); //True
console.log(isMatch("ab", ".*")); //True

File: Hard/regular_expression_matching.go
package main

import (
	"fmt"
	"regexp"
)

func isMatch(s string, p string) bool {
	regx := fmt.Sprintf("^%s$", p)
	isMatch, err := regexp.MatchString(regx, s)
	if err != nil {
		return false
	}

	return isMatch
}

func run_isMatch() {
	fmt.Println(isMatch("aa", "a"))  //False
	fmt.Println(isMatch("aa", "a*")) //True
	fmt.Println(isMatch("ab", ".*")) //True
}
